### PR TITLE
[fix bug 1392796] Remove old Fx desktop & private browsing pages.

### PR DESCRIPTION
**Pull request by jpetto**
## Description

Removes some pages deprecated by opening up new Fx hub to all locales:

- `/firefox/desktop/fast/`
- `/firefox/desktop/trust/`
- `/firefox/desktop/tips/`
- `/firefox/desktop/customize/`
- `/firefox/private-browsing/`

~~**DO NOT MERGE** while I verify the redirect for `/firefox/desktop/customize/`. Should be good for review though.~~

I'm going to say the redirect for `/firefox/desktop/customize/` is validated by the filing of [bug 1393881](https://bugzilla.mozilla.org/show_bug.cgi?id=1393881).

## Issue / Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1392796

## Testing

Ensure no regressions and that all associated files have been removed. Make sure appropriate redirects are in place.

Integration tests passing: https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/51/pipeline/190/

### CODE REVIEW

File: bedrock/firefox/redirects.py [hunk 1]
```diff
@@ -145,9 +145,10 @@ def firefox_channel(*args, **kwargs):
     redirect(r'^firefox/fx/?$', 'firefox.new'),
 
     # Bug 979531, 1003727, 979664, 979654, 979660
-    redirect(r'^firefox/customize/?$', 'firefox.desktop.customize'),
-    redirect(r'^firefox/(?:performance|happy|speed|memory)/?$', 'firefox.desktop.fast'),
-    redirect(r'^firefox/security/?$', 'firefox.desktop.trust'),
+    # TODO: verify URL below
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py]: Noting TODO

File: bedrock/firefox/redirects.py [hunk 2]
```diff
@@ -540,4 +541,12 @@ def firefox_channel(*args, **kwargs):
 
     # bug 1386112
     redirect(r'^firefox/android/faq/?', 'https://support.mozilla.org/products/mobile'),
+
+    # bug 1392796
+    redirect(r'^firefox/desktop/fast/?', 'firefox.features.fast'),
+    redirect(r'^firefox/desktop/trust/?', 'firefox.features.independent'),
+    redirect(r'^firefox/desktop/tips/?', 'firefox.features.index'),
+    # TODO: verify link below
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py]: Noting TODO

File: tests/redirects/map_globalconf.py [hunk 3]
```diff
@@ -335,9 +335,10 @@
     url_test('/firefox/fx/', '/firefox/new/'),
 
     # Bug 979531, 1003727, 979664, 979654, 979660, 1150713
-    url_test('/firefox/customize/', '/firefox/desktop/customize/'),
-    url_test('/firefox/{performance,happy,speed,memory}/', '/firefox/desktop/fast/'),
-    url_test('/firefox/security/', '/firefox/desktop/trust/'),
+    # TODO: verify URL below
```
Comment (alexgibson) [inline tests/redirects/map_globalconf.py]: Noting TODO

File: tests/redirects/map_globalconf.py [hunk 4]
```diff
@@ -1165,4 +1166,11 @@
 
     # Bug 1386112
     url_test('/firefox/android/faq/', 'https://support.mozilla.org/products/mobile'),
+
+    # bug 1392796
+    url_test('/firefox/desktop/trust/', '/firefox/features/independent/'),
+    url_test('/firefox/desktop/tips/', '/firefox/features/'),
+    # TODO: verify URL below
```
Comment (alexgibson) [inline tests/redirects/map_globalconf.py]: Noting TODO

File: bedrock/firefox/urls.py [hunk 5]
```diff
@@ -79,9 +73,6 @@
     url(r'^firefox/ios/testflight', views.ios_testflight, name='firefox.ios.testflight'),
     page('firefox/mobile-download', 'firefox/mobile-download.html'),
     page('firefox/mobile-download/desktop', 'firefox/mobile-download-desktop.html'),
-    page('firefox/private-browsing', 'firefox/private-browsing.html', alt_canonical_paths={
```
Comment (alexgibson) [inline bedrock/firefox/urls.py]: Searching through the repo I can still find several references to `{{ url('firefox.private-browsing') }}` which will need updating.

File: tests/pages/firefox/desktop/customize.py [hunk 6]
```diff
@@ -1,87 +0,0 @@
-# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (alexgibson) [inline tests/pages/firefox/desktop/customize.py]: I can still see references to this file in `tests/functional/newsletter/test_newsletter_embed.py`

File: bedrock/firefox/urls.py [hunk 7]
```diff
@@ -36,12 +36,6 @@
     url(r'^firefox/desktop/$',
         views.FirefoxProductDesktopView.as_view(),
         name='firefox.desktop.index'),
-    page('firefox/desktop/fast', 'firefox/desktop/fast.html', alt_canonical_paths={
-        'en-US': '/firefox/features/fast/',
-    }),
-    page('firefox/desktop/customize', 'firefox/desktop/customize.html'),
-    page('firefox/desktop/tips', 'firefox/desktop/tips.html'),
-    page('firefox/desktop/trust', 'firefox/desktop/trust.html'),
```
Comment (alexgibson) [inline bedrock/firefox/urls.py]: There's still a reference to `url('firefox.desktop.trust')` in `firefox/dnt.html`

File: bedrock/firefox/templates/firefox/desktop/desktop-base.html [hunk 8]
```diff
@@ -1,37 +0,0 @@
...
-{% extends "firefox/base-resp.html" %}
-
-{% block page_title_prefix %}{% endblock %}
-{% block page_title %}{% endblock %}
-
-{% block body_id %}firefox-desktop{% endblock %}
-{% block body_class %}firefox-desktop{% endblock %}
-
-{% block extrahead %}
-  <script src="{{ static('js/libs/modernizr.custom.cssanimations.js') }}"></script>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/desktop/desktop-base.html]: Can this JS file be removed? Seems like we can just use `@supports` these days?

Comment (jpetto) [reply bedrock/firefox/templates/firefox/desktop/desktop-base.html]: Ah, good catch. Looks like there's another orphaned Modernizr file in there, too.

File: tests/pages/firefox/desktop/customize.py (hunk 6 above)
Comment (jpetto) [reply tests/pages/firefox/desktop/customize.py]: Is it weird that tests passed even with this outdated reference? Anyway, nice spot - will remove.

File: bedrock/firefox/urls.py (hunk 5 above)
Comment (jpetto) [reply bedrock/firefox/urls.py]: Got 'em.

File: bedrock/firefox/urls.py (hunk 7 above)
Comment (jpetto) [reply bedrock/firefox/urls.py]: Bah, nice spot. 🕵️

File: tests/pages/firefox/desktop/customize.py (hunk 6 above)
Comment (alexgibson) [reply tests/pages/firefox/desktop/customize.py]: Where did tests pass?

Comment (jpetto) [reply tests/pages/firefox/desktop/customize.py]: Apparently I forgot functional tests do not run on PRs.

File: bedrock/firefox/redirects.py [hunk 9]
```diff
@@ -540,4 +540,11 @@ def firefox_channel(*args, **kwargs):
 
     # bug 1386112
     redirect(r'^firefox/android/faq/?', 'https://support.mozilla.org/products/mobile'),
+
+    # bug 1392796
+    redirect(r'^firefox/desktop/fast/?', 'firefox.features.fast'),
+    redirect(r'^firefox/desktop/trust/?', 'firefox.features.independent'),
+    redirect(r'^firefox/desktop/tips/?', 'firefox.features.index'),
+    redirect(r'^firefox/desktop/customize/?', 'https://support.mozilla.org/kb/customize-firefox-controls-buttons-and-toolbars'),
+    redirect(r'^firefox/private-browsing/?', 'firefox.features.private-browsing'),
```
Comment (alexgibson) [inline bedrock/firefox/redirects.py]: Do we need to remove the existing redirects for `/en-US/firefox/private-browsing/` above on line 525, and `/en-US/firefox/desktop/fast/` on line 529 above now that we have these newer all-locale redirects? We should probably update the redirect tests to reflect these changes too.

Comment (jpetto) [reply bedrock/firefox/redirects.py]: Nice spot. I think we're fine just removing the `en-US` only redirects and their tests, as the new redirects/tests cover all locales.

Comment (alexgibson) [reply bedrock/firefox/redirects.py]: Are we missing a redirect test for `/firefox/fast/`?

Comment (jpetto) [reply bedrock/firefox/redirects.py]: > Are we missing a redirect test for /firefox/fast/?

Was that ever a live URL? 404's on prod currently. There is a redirect & test for `/firefox/desktop/fast/`...

Comment (alexgibson) [reply bedrock/firefox/redirects.py]: Sorry I meant `/firefox/features/fast/` I can only see a test for the en-US redirect

Comment (jpetto) [reply bedrock/firefox/redirects.py]: I have that test locally - waiting for integration tests to finish/pass before updating the PR. 😉

**Comment by jpetto** [comment]
Latest passing integration test run: https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/52/pipeline/

**Pull request merged**